pkg/routeagent_driver/cni/cni_iface.go

```diff
@@ -86,9 +86,20 @@ func discover(clusterCIDR string) (*Interface, error) {
 }
 
 func AnnotateNodeWithCNIInterfaceIP(nodeName string, clientSet kubernetes.Interface, clusterCidr []string) error {
-	cniIface, err := Discover(clusterCidr[0])
-	if err != nil {
-		return errors.Wrap(err, "Discover returned error")
+	cniIPAddress := ""
+	setAnnotation := true
+
+	if len(clusterCidr) == 0 {
+		setAnnotation = false
+	}
+
+	if setAnnotation {
+		cniIface, err := Discover(clusterCidr[0])
+		if err != nil {
+			return errors.Wrap(err, "Discover returned error")
```

Comment:
I would suggest an error such as "error retrieving the CNI interface for %s: %v", clusterCidr[0], err instead.